spring-core/src/main/java/org/springframework/cglib/core/ReflectUtils.java

```diff
@@ -382,7 +382,7 @@ public static Method findNewInstance(Class iface) {
 			}
 		}
 		methods.remove(null);
-		return (Method[]) methods.toArray(new Method[methods.size()]);
+		return (Method[]) methods.toArray(new Method[0]);
```

Comment:
This is bad, because it it creates **2 arrays**: first with `new Method[0]`, then `toArray()` will see that the capacity of the given array is not sufficient and will create **one more array**.

Reply:
I agree with @mentallurg here.

Please revert this change.

Reply:
What about https://shipilev.net/blog/2016/arrays-wisdom-ancients/ ?

Reply:
@andrei-ivanov The information you provided is much more detailed than mine. I have learned a lot from it, thank you.

Reply:
Indeed. I had momentarily forgotten about those findings from Shipilёv.

Thanks for the links.

We'll keep the proposed change! 👍